internal/pkg/scaffold/role.go

```diff
@@ -229,5 +229,17 @@ rules:
   - {{ .ProjectName }}
   verbs:
   - "update"
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+- apiGroups:
+  - apps
+  resources:
+  - replicasets
```

Comment:
Why `replicasets` here?

Reply:
This is necessary for code that moves up the owner reference chain from the pod to figure out which resource to use for the metrics service owner reference. Since our default operator workload kind is `Deployment`, the code needs to be able to `GET` both `pods` and `replicasets`. Since we end up using the `replicaset`'s owner reference, we don't actually need `GET` permission on `deployments`.

Reply:
We already have pods and replicasets listed with the verb `*` above in Lines 164-184
```YAML
- apiGroups:
  - ""
  resources:
  - pods
...
  verbs:
  - "*"
- apiGroups:
  - apps
  resources:
  - deployments
  - daemonsets
  - replicasets
  - statefulsets
  verbs:
  - "*"
```

Those are the general default rules and these ones are specific to allowing metrics to be exposed. Do you think it's worth adding a comment in the scaffold so it's obvious to the user why we've repeated the rules for pods and replicasets?

Reply:
This would be nice. Lili asked this same question when we introduced the initial helm role generation. I looked into it and it turns out [it isn't straigtforward](https://github.com/operator-framework/operator-sdk/pull/1188#discussion_r270907519) since we unmarshal and remarshal the YAML (thus losing the comments) to update the role for CRD APIs.